src/NuGet.Core/NuGet.Configuration/PackageNamespaces/SearchNode.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace NuGet.Configuration
+{
+    internal class SearchNode
+    {
+        public readonly SearchNode Parent;
+        public readonly Dictionary<char, SearchNode> Children;
+        public bool IsValueNode => !string.IsNullOrWhiteSpace(NamespaceId);
+        public bool IsGlobbing { get; set; }
+        public string NamespaceId { get; set; }
```

Comment:
If we are building the `Trie` datastructure for `NamespaceId(s)`, do we need to maintain the `NamespaceId` on every node?

Reply:
Actual trie I have seen either keep actual `string` value for each node or use another boolean flag. Currently it's internal type such small details are not important. I'll do another iteration of clean up for perf optimization once everything is more clear.

Reply:
Created follow up issue for optimization. We can do compaction for once trie tree building is complete. https://github.com/NuGet/Client.Engineering/issues/985